src/thunderbots/software/radio_communication/mrf_backend.cpp

```diff
@@ -1,18 +1,100 @@
+     * the build IDs.
+     */
+    const double REQUEST_BUILD_IDS_INTERVAL = 0.5;
+
+    struct RSSITableEntry final
+    {
+        int rssi;
+        int db;
+    };
+    const struct RSSITableEntry RSSI_TABLE[] = {
+        {255, -35}, {254, -36}, {253, -37}, {250, -38}, {245, -39}, {239, -40},
+        {233, -41}, {228, -42}, {225, -43}, {221, -44}, {216, -45}, {212, -46},
+        {207, -47}, {203, -48}, {198, -49}, {193, -50}, {188, -51}, {183, -52},
+        {176, -53}, {170, -54}, {165, -55}, {159, -56}, {153, -57}, {148, -58},
+        {143, -59}, {138, -60}, {133, -61}, {129, -62}, {125, -63}, {121, -64},
+        {117, -65}, {111, -66}, {107, -67}, {100, -68}, {95, -69},  {89, -70},
+        {83, -71},  {78, -72},  {73, -73},  {68, -74},  {63, -75},  {58, -76},
+        {53, -77},  {48, -78},  {43, -79},  {37, -80},  {32, -81},  {27, -82},
+        {23, -83},  {18, -84},  {13, -85},  {9, -86},   {5, -87},   {2, -88},
+        {1, -89},   {0, -90},
+    };
+
+    const char *const SD_MESSAGES[] = {
+        nullptr,
+        u8"Bot %1 SD card uninitialized",
+        nullptr,
+        u8"Bot %1 SD card incompatible",
+        u8"Bot %1 SD card sent illegal response",
+        u8"Bot %1 SD layer logical error",
+        u8"Bot %1 SD card CRC error",
+        u8"Bot %1 SD card claimed illegal command",
+        u8"Bot %1 SD card in unexpected state",
+        u8"Bot %1 SD card internal error",
+        u8"Bot %1 SD card command response timeout",
+        u8"Bot %1 SD card parameter out of range",
+        u8"Bot %1 SD card address misaligned",
+        u8"Bot %1 SD card block length error",
+        u8"Bot %1 SD card erase sequence error",
+        u8"Bot %1 SD card erase parameter error",
+        u8"Bot %1 SD card write protect violation",
+        u8"Bot %1 SD card locked",
+        u8"Bot %1 SD card lock or unlock failed",
+        u8"Bot %1 SD card command CRC error",
+        u8"Bot %1 SD card ECC error",
+        u8"Bot %1 SD card CC error",
+        u8"Bot %1 SD card generic error",
+        u8"Bot %1 SD card CSD write error",
+        u8"Bot %1 SD card partial erase due to write protection",
+        u8"Bot %1 SD card ECC disabled",
+        u8"Bot %1 SD card erase sequence cancelled",
+        u8"Bot %1 SD card authentication sequence error",
+        u8"Bot %1 SD card initialization timeout",
+        u8"Bot %1 SD card data timeout",
+        u8"Bot %1 SD card data CRC error",
+        u8"Bot %1 SD card missing data start bit",
+        u8"Bot %1 SD card FIFO overrun or underrun",
+    };
+
+    const char *const LOGGER_MESSAGES[] = {
+        nullptr,
+        u8"Bot %1 logger uninitialized",
+        nullptr,
+        u8"Bot %1 SD card full",
+    };
+}  // namespace
+
+MRFBackend::MRFBackend(ros::NodeHandle &node_handle)
+    : dongle(MRFDongle(std::bind(&MRFBackend::handle_message, this, ph::_1, ph::_2,
+                                 ph::_3, ph::_4, ph::_5))),
```

Comment:
scary bind is scary, let's add a quick comment roughly explaining what this does.....

Reply:
removed, not relevant anymore

Reply:
my fav. soln